part1/src/main.rs

```diff
@@ -16,16 +16,21 @@ struct Interpreter {
 }
 
 impl Interpreter {
-    pub fn new(input: String) -> Self {
+    pub fn new(input: &str) -> Self {
         Interpreter {
-            input,
+            input: input.to_string(),
```

Comment:
this is because of whitespace right?

Reply:
No, it's because I changed the `input` type from `String` to `&str` to make testing easier.

This allows test functions to make calls like
```rust
Interpreter::new("1+2")
```
instead of
```rust
Interpreter::new("1+2".to_string())
```

Reply:
Yea but changing code just for tests? seems like it doesn't justify the code change

Reply:
Assuming it doesn't make the non-test interface *worse*, I do advocate for changes to the production code that improve testability. In this case, making `Interpreter::new` take a `&str` rather than `String` seems fine to me. Having a bunch of `to_string()`s in your test code is going to make it harder to see what the code is actually testing.